marimo/_server/export/exporter.py

```diff
@@ -118,16 +121,41 @@ def export_as_script(
             graph.cells[cid].code
             for cid in dataflow.topological_sort(graph, graph.cells.keys())
         ]
-        filename = file_manager.filename
-        if not filename:
-            filename = "notebook.script.py"
-
-        basename = os.path.basename(filename)
-        download_filename = f"{os.path.splitext(basename)[0]}.script.py"
-
         code = (
             f'\n__generated_with = "{__version__}"\n\n'
             + "\n\n# ---\n\n".join(codes)
         )
 
+        download_filename = get_download_filename(file_manager, ".script.py")
         return code, download_filename
+
+    def export_as_md(self, file_manager: AppFileManager) -> tuple[str, str]:
+        # TODO: Provide filter or kernel in header such that markdown documents
+        # are executable.
+        document = [
+            dedent(
+                f"""
+          ---
+          title: {get_filename_title(file_manager)}
+          marimo-version: {__version__}
+          ---"""
+            ).strip(),
+            "",
+        ]
+        for cell_data in file_manager.app.cell_manager.cell_data():
+            cell = cell_data.cell
+            code = cell_data.code
+            if cell:
+                markdown = get_markdown_from_cell(cell, code)
+                if markdown:
+                    document.append(markdown)
+                else:
+                    guard = "```"
+                    while guard in code:
+                        guard += "`"
+                    document.extend(
+                        [f"""{guard}{{marimo}}""", code, guard, ""]
```

Comment:
im not sure how these tags work in quarto or other platforms, but is it possible to get python syntax highlighting too, with the codefence being `python {marimo}`

```python {marimo}
def foo():
    return x
```

Reply:
I checked in VSCode and "{.python.marimo}" natively highlights.

Think {.*marimo.*} Is too loose converting back?